platforms/core-configuration/kotlin-dsl-tooling-builders/build.gradle.kts

```diff
@@ -5,19 +5,19 @@ plugins {
 description = "Kotlin DSL Tooling Builders for IDEs"
 
 dependencies {
-    implementation(project(":kotlin-dsl"))
+    api(project(":core-api"))
+    api(project(":core"))
+    api(libs.futureKotlin("stdlib"))
 
+    implementation(project(":kotlin-dsl"))
     implementation(project(":base-services"))
-    implementation(project(":core-api"))
-    implementation(project(":model-core"))
-    implementation(project(":core"))
     implementation(project(":resources"))
     implementation(project(":platform-base"))
     implementation(project(":platform-jvm"))
-    implementation(project(":plugins-java"))
     implementation(project(":plugins-java-base"))
     implementation(project(":tooling-api"))
     implementation(project(":logging"))
+    implementation(project(":kotlin-dsl-tooling-models"))
```

Comment:
❌  Why do you need to introduce dependencies for the main source set when your change only affects tests?